webapp/publisher/views.py

```diff
@@ -78,16 +80,55 @@ def get_account_details():
         return _handle_errors(api_error)
 
     flask_user = flask.session["openid"]
+
+    subscriptions = None
+    subscribed_to_newsletter = None
+    # don't rely on marketo to show the page,
+    # if anything fails, just continue and don't show
+    # this section
+    try:
+        marketo_user = marketo.get_user(flask_user["email"])
+        marketo_subscribed = marketo.get_newsletter_subscription(
+            marketo_user["id"]
+        )
+        subscribed_to_newsletter = False
+        if (
+            "snapcraftnewsletter" in marketo_subscribed
```

Comment:
I think this could also be just `if marketo_subscribed.get("snapcraftnewsletter")`.